core/src/main/java/org/jdbi/v3/core/cache/internal/DoubleLinkedList.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.cache.internal;
+
+final class DoubleLinkedList<K, V> {
```

Comment:
How is this DoubleLinkedList implementation better than out of the box LinkedList, which is already doubly linked?

Reply:
Ah, I suppose it lets you remove nodes rather than elements. Too bad we have to reimplement the whole data structure.

Reply:
yes. To be able to implement LRU effectively with a DLL, you need to be able to unlink and reorder the nodes. The LinkedList implementation does not allow that.